cranelift/codegen/src/opts/arithmetic.isle

```diff
@@ -347,3 +347,16 @@
 ;; (x - y) - x => -y
 (rule (simplify (isub ty (isub ty x y) x))(ineg ty y))
 
+;; (x * C) (==/!=) D --> x (==/!=) (D / C) when C is odd and divides D
+(rule (simplify 
+	(ne cty (imul ty x (iconst_u ty y)) (iconst_u ty z)))
```

Comment:
One more request, sorry -- could you add the commutative versions (const * x) for both rules as well?